source/MaterialXRender/Handlers/Mesh.cpp

```diff
@@ -41,46 +48,55 @@ bool MeshPartition::generateTangents(MeshStreamPtr positionStream, MeshStreamPtr
     const unsigned int tangentStride = 3;
     tangentStream->setStride(tangentStride);
 
-    for (size_t faceIndex = 0; faceIndex < getFaceCount(); faceIndex++)
+    for (size_t i = 0; i < getPartitionCount(); i++)
```

Comment:
Hopefully not to hard to compare. I basically moved this to the mesh level as it trying to set tangents across the entire mesh per partition erroneously.